src/TimerMainDialog.cpp

```diff
@@ -0,0 +1,379 @@
+   timeLCD->display(timeToString(time));
+}
+
+QString TimerMainDialog::timeToString(int t)
+{
+    unsigned int seconds = t;
+    unsigned int minutes = 0;
+    unsigned int hours = 0;
+   if (t == 0)
+       return "00:00:00";
+   if (t > 59){
+       seconds = t%60;
+       minutes = t/60;
+       if (minutes > 59){
+          hours = minutes/60;
+          minutes = minutes%60;
+       }
+   }
+   QString secStr, minStr, hourStr;
+   if (seconds < 10)
+       secStr = "0" + QString::number(seconds);
+   else
+       secStr = QString::number(seconds);
+   if (minutes <10)
+       minStr = "0" + QString::number(minutes);
+   else
+       minStr = QString::number(minutes);
+   if (hours < 10)
+       hourStr = "0" + QString::number(hours);
+   else
+       hourStr = QString::number(hours);
+   return hourStr + ":" + minStr + ":" + secStr;
+}
+
+
+void TimerMainDialog::on_hideButton_clicked()
+{
+    hideTimers();
+}
+
+void TimerMainDialog::hideTimers()
+{
+    if (!timerWindow->isHidden())
+        timerWindow->hide();
+}
+
+void TimerMainDialog::on_showButton_clicked()
+{
+    if (!timers->isEmpty()) {
+        if (timerWindow->isHidden())
+            timerWindow->show();
+    }
+    else
+            QMessageBox::warning(this, "No Timers", "There are currently no timers to show.");
+}
+
+void TimerMainDialog::timesUp()
+{
+    //If there are no knockout timers generate a timer for this
+    if (!stopped) {
+        bool isKnockOutTimer = false;
+        QString note = "KNOCKOUT";
+        foreach (TimerWidget* t, *timers) {
+           if (t->getTime() == 0) {
+               isKnockOutTimer = true;
+               t->setNote(note); //update existing timers note
+            }
+        }
+        if (!isKnockOutTimer)
+            createTimer(note);
+        stopped = true;
+    }
+}
+
+void TimerMainDialog::on_loadRecipesButton_clicked()
+{
+    //Load current recipes
+    if (!timers->isEmpty()){
+        QMessageBox mb;
+        mb.setText("Active Timers");
+        mb.setInformativeText("You currently have active timers, would you like to replace them or add to them?");
+        QAbstractButton *replace =  mb.addButton(tr("Replace"), QMessageBox::YesRole);
+        QAbstractButton *add = mb.addButton(tr("Add"), QMessageBox::NoRole);
+        add->setFocus();
+        mb.setIcon(QMessageBox::Question);
+        mb.exec();
+        if (mb.clickedButton() == replace)
+            removeAllTimers();
+    }
+    Recipe* recipe = mainWindow->currentRecipe();
+    setBoilTimeBox->setValue(recipe->boilTime_min());
+    bool timerFound = false;
+    int duplicates = 0;
+    int timersGenerated = 0;
+    bool duplicatesFound = false;
+    QList<Hop*> hops;
+    QString note;
+    hops = recipe->hops();
+    foreach (Hop* h, hops) {
+        if (h->use() == 2) { //2 = Boil addition -- Hop::Use enum
+            note = Brewtarget::displayAmount(h->amount_kg(), "TimerNote", "hop_amount", Units::kilograms) +
```

Comment:
Make sure all user-visible strings are translatable with [QObject::tr()](http://doc.qt.io/qt-5/qobject.html#tr). For example, this one should be

    note = tr("%1 of %2").arg(Brewtarget::displayAmount(...)).arg(h->name());